### PR TITLE
Enhance: Full Connection Screen

### PR DESCRIPTION
**Pull request by jeffgolenski**
Creating a full connection screen on the plugin page. 

Fixes https://github.com/Automattic/jetpack/issues/6067

#### Changes proposed in this Pull Request:

Next up is a full screen connection dialog that will display immediately after activation of Jetpack (on the plugins page.) This will display only immediately after JP is activated on this screen, and we'll want to test it. If Jetpack is somehow already installed and activated in an install, the 3 aforementioned methods will be backup for ensuring user connection to wpcom.

Design (in browser):
screen shot 2017-03-10 at 12 11 34 pm

full-connection-screen-mobile

#### To do:

Some logic still needs to be built in for this to work properly.

- [x] Dismiss "X" needs to dismiss the full-screen connection prompt. We *don't want to deactivate* Jetpack when this banner is dismissed—we want to display the connection banner at the top of the plugins list: https://cloudup.com/cRBS7jLJCWa
- [x] Logic needs to be written to display this banner *only* when Jetpack is installed + activated. This prompt should be displayed immediately after plugin activation. If they dismiss this full screen prompt, the user should never see it again. 
- [x] We need to track some data! Can we get some tracks on the dismissal "X" and the connection button? cc @dereksmart (since you're my go-to guy for tracks implementation)

#### Testing instructions:

1) Disconnect Jetpack
2) Visit /wp-admin/plugins/
3) Bask in the glory of the new full screen connection prompt

### CODE REVIEW

**Comment by dereksmart** [comment]
> Can we get some tracks on the dismissal "X"

Unfortunately we can't send back any data on this until they've connected.  However, we have a way to track how many click to connect.

**Comment by dereksmart** [comment]
@jeffgolenski I've pushed some updates to this PR which includes: 

- Only showing the prompt once, and only after activation.  Technically, if the user lands on this page within 6 seconds after activating the plugin, they will see it.  I can make this time period shorter or longer. 
- The prompt can be dismissed by clicking the Dismiss, or with an `esc` keypress.  
- Currently we're not remembering if they've dismissed it, so if they dismiss the prompt, then only if they deactivate and reactivate the plugin they will see the notice again.  I can make dismissal completely permanent if you think it's best.  
- As I commented above, we can't track dismissals, however we are now sending a new prop to track for `jetpack_connection_view` events.  The prop name is `full-screen-prompt`

It's ready for testing/review.

**Comment by jeffgolenski** [comment]
@dereksmart This is looking fantastic. The functionality is right on point. Thanks for getting tracks into this as well! 

jetpack-full-banner

(don't mind all the colors being weird in this .gif, it's just a compression thing)

**Comment by jeffgolenski** [comment]
I've just got to make an adjustment to the dismiss "x" which currently disappears on small screens. Then I think we'll be good to go!

**Comment by jeffgolenski** [comment]
Alright we're gold! Let's get this in the next release!

File: class.jetpack-connection-banner.php [hunk 1]
```diff
@@ -463,6 +470,73 @@ class="dops-button is-primary">
...
+						?>
+					</p>
+				</div>
+				<div class="jp-connect-full__card-footer">
+					<p class="jp-connect-full__tos-blurb">
+						<?php
+						printf(
+							__( 'By connecting your site you agree to our fascinating <a href="%s" target="_blank" class="jp-connect-full__tos-a">Terms of Service</a> and to <a href="%s" target="_blank" class="jp-connect-full__tos-a">share details</a> with WordPress.com', 'jetpack' ),
+							esc_url( 'https://wordpress.com/tos' ),
+							esc_url( 'https://jetpack.com/support/what-data-does-jetpack-sync' )
```
Comment (eliorivero) [inline class.jetpack-connection-banner.php]: These two URLs don't need to be escaped since there's no dynamic data here.

**Pull request merged**